op-node/sources/l2_client.go

```diff
@@ -0,0 +1,124 @@
+package sources
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ethereum-optimism/optimism/op-node/client"
+	"github.com/ethereum-optimism/optimism/op-node/eth"
+	"github.com/ethereum-optimism/optimism/op-node/rollup"
+	"github.com/ethereum-optimism/optimism/op-node/rollup/derive"
+	"github.com/ethereum-optimism/optimism/op-node/sources/caching"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+type L2ClientConfig struct {
+	EthClientConfig
+
+	L2BlockRefsCacheSize int
+
+	Genesis rollup.Genesis
+}
+
+func L2ClientDefaultConfig(config *rollup.Config, trustRPC bool) *L2ClientConfig {
+	// Cache 3/2 worth of sequencing window of payloads, block references, receipts and txs
+	span := int(config.SeqWindowSize) * 3 / 2
+	// Estimate number of L2 blocks in this span of L1 blocks
+	// (there's always one L2 block per L1 block, L1 is thus the minimum, even if block time is very high)
+	if config.BlockTime < 12 && config.BlockTime > 0 {
```

Comment:
I don't think its a bad idea to add some loglines here, geth does this when sanitizing config. A bug resulting from silent modifications of config can be really annoying

Reply:
It's not modifying any config value, it's calculating a new one. Right now this default-config-generating function is a pure function without logging/error edge cases, And no changes to the input config, just a new output. So I think it's fine.